Lib/fontTools/pens/hashPointPen.py

```diff
@@ -0,0 +1,58 @@
+# Modified from https://github.com/adobe-type-tools/psautohint/blob/08b346865710ed3c172f1eb581d6ef243b203f99/python/psautohint/ufoFont.py#L800-L838
+import hashlib
+
+from fontTools.misc.transform import Identity
+from fontTools.pens.pointPen import AbstractPointPen
+
+
+class HashPointPen(AbstractPointPen):
+    """
+    This pen can be used to check if a glyph's contents (outlines plus
+    components) have changed.
+
+    Components are decomposed, because that is what happens in TrueType
```

Comment:
the docstring here no longer reflects what the pen is doing with components, could you please update this as well?

Reply:
Sure!